### DISCUSSION TITLE
How can I adverise and publish custom ros message via adverise API

### DISCUSSION THREAD

**Discussion by hql420621** [foxglove/community#6] [Q&A]
hello, recently, i am struggle with how to publish custom ros message via extension API, but it doenst work ,could you tell me how to use it properly .
```
context.advertise?.("/my_parking_topic", "cm_node/test")
context.publish?.("/my_parking_topic", {
      smDrvrAsscSysBtnPush: 10,
      smDrvrAsscSysBtn: 20,
    });

`cm_node/test` is my custom datetype ,looks like below
```
type TestType = {
  smDrvrAsscSysBtnPush: number;
  smDrvrAsscSysBtn: number;
}

but it seems the foxglove cannot  parse custom defined data type ,is there a exampe to correctly use `advertise` API  to publish custom data message ,thanks in advance

**Comment by defunctzombie** [comment]
Are you using Rosbridge for your connection?

I also see you are using optional chaining on your call to advertise: `context.advertise?.`. If you are not connect to a data source that has publish capability then this call will not do anything since `context.advertise` will be undefined. Have you made sure that the `advertise` function is defined before you call it?

**Reply by hql420621** [reply]
thanks for answering,yes,i use Rosbridge for my connection, i have solve it 
i seems i have to source the ros project in the same console with roslaunch, the the rosbridge can identify my custom datatype
